spec/rspec/core/example_group_spec.rb

```diff
@@ -1372,6 +1373,37 @@ def extract_execution_results(group)
         end
       end
 
+      context "with fail_fast and failures_required = 3" do
+        let(:group) do
+          group = RSpec.describe
+          allow(group).to receive(:fail_fast?) { true }
+          allow(group).to receive(:failures_required) { 3 }
+          group
+        end
+        let(:config)   { Configuration.new }
+        let(:reporter) { Reporter.new config }
+
+        it "does not run examples after 3 failed examples" do
+          examples_run = []
+          self.group.example('example 1') { examples_run << self }
+          self.group.example('example 2') { examples_run << self; fail; }
+          self.group.example('example 3') { examples_run << self; fail; }
+          self.group.example('example 4') { examples_run << self; fail; }
+          self.group.example('example 5') { examples_run << self }
```

Comment:
You don't need `self.` here... rename the local instead to avoid the ruby 2.2 warning